### PR TITLE
Add: OSS::Skeleton component

### PR DESCRIPTION
**Pull request by aprentout**
### What does this PR do?
Add new component OSS::Skeleton

Related to: [#ENG-510](https://linear.app/upfluence/issue/ENG-510/ossskeleton)

### What are the observable changes?
Capture d’écran 2023-07-05 à 16 34 01

https://github.com/upfluence/oss-components/assets/111493996/6ec8cf16-e6f4-4c65-9432-a3cbada69469

### Good PR checklist
- [x] Title makes sense
- [x] Is against the correct branch
- [x] Only addresses one issue
- [x] Properly assigned
- [x] Added/updated tests
- [x] Added/updated documentation
- [ ] Migrated touched components to Glimmer Components
- [x] Properly labeled

### CODE REVIEW

File: addon/components/o-s-s/skeleton.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+interface Row {
+  style?: string;
+}
+
+interface OSSSkeletonArgs {
+  width: number;
+  height: number;
+  multiple?: number;
+  type?: 'row' | 'column';
+  gap?: number;
```
Comment (Miexil) [inline addon/components/o-s-s/skeleton.ts:13]: Can we also put the randomization as a parameter please ? 😇

File: addon/components/o-s-s/skeleton.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+export default class OSSSkeleton extends Component<OSSSkeletonArgs> {
+  constructor(owner: unknown, args: OSSSkeletonArgs) {
+    super(owner, args);
+
+    if (this.args.type) {
+      assert(
+        `[component][OSS::Skeleton] The @type argument should be a value of ${['row', 'column']}`,
+        ['row', 'column'].includes(this.args.type)
+      );
+    }
```
Comment (Miexil) [inline addon/components/o-s-s/skeleton.ts:25]: maybe assert width & height as well as they are not optional values ?

Comment (Miexil) [reply addon/components/o-s-s/skeleton.ts:25]: Ah, I see there are default values below, maybe tag them as optional in the OSSSkeletonArgs interface then ?

File: addon/components/o-s-s/skeleton.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    } else {
+      for (let i = 0; i < multiple; i++) {
+        rows.push(this._generateRandomRowSize());
+      }
+    }
+
+    return rows;
+  }
+
+  _generateRandomRowSize(): Row {
```
Comment (Miexil) [inline addon/components/o-s-s/skeleton.ts:74]: remove `_` add `private` keyword instead

File: addon/components/o-s-s/skeleton.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    if (this.args.type) {
+      assert(
+        `[component][OSS::Skeleton] The @type argument should be a value of ${['row', 'column']}`,
+        ['row', 'column'].includes(this.args.type)
+      );
+    }
+  }
+
+  get height(): number {
+    return this.args.height || 100;
```
Comment (Miexil) [inline addon/components/o-s-s/skeleton.ts:29]: default height should be something more reasonable, maybe `36px` to match  our input height ?

File: tests/integration/components/o-s-s/skeleton-test.ts [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+  });
+
+  module('@height parameters', () => {
+    test('Default height', async function (assert) {
+      await render(hbs`<OSS::Skeleton/>`);
+
+      assert.dom('.upf-skeleton-content').hasStyle({ height: '100px' });
+    });
+
+    test('Then style height should correspond to parameter value', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/skeleton-test.ts:21]: The* ?

File: tests/integration/components/o-s-s/skeleton-test.ts [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  });
+
+  module('@width parameters', () => {
+    test('Default width', async function (assert) {
+      await render(hbs`<OSS::Skeleton/>`);
+
+      assert.dom('.upf-skeleton-content').hasStyle({ width: '100px' });
+    });
+
+    test('Then style width should correspond to parameter value', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/skeleton-test.ts:37]: The* ?

File: tests/integration/components/o-s-s/skeleton-test.ts [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+  module('@multiple parameters', () => {
+    test('Default has one skeleton effect', async function (assert) {
+      await render(hbs`<OSS::Skeleton/>`);
+
+      let items = findAll('.upf-skeleton-content .upf-skeleton-effect');
+
+      assert.ok(items.length === 1);
+    });
+
+    test('Then has multiple skeleton effect', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/skeleton-test.ts:71]: what does `then` refer to here ? 🤔

File: tests/integration/components/o-s-s/skeleton-test.ts [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    });
+  });
+
+  module('When type is row', () => {
+    hooks.beforeEach(function () {
+      this.multiple = 4;
+      this.width = 200;
+    });
+
+    test('Then content should have class fx-col', async function (assert) {
```
Comment (Miexil) [inline tests/integration/components/o-s-s/skeleton-test.ts:88]: Ah, I think I understand the `then`s I've commented above.
`Then` makes sense if there is a `when` in the module. here you should add `Then the content`

File: addon/components/o-s-s/skeleton.stories.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+      table: {
+        type: {
+          summary: 'number'
+        },
+        defaultValue: { summary: '100' }
+      },
+      control: { type: 'number' }
+    },
+    multiple: {
+      description: 'How many skeleton effect should be displayed',
```
Comment (Miexil) [inline addon/components/o-s-s/skeleton.stories.js:30]: effects*
or rows

File: addon/components/o-s-s/skeleton.stories.js [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        defaultValue: { summary: 'row' }
+      },
+      options: DirectionTypes,
+      control: { type: 'select' }
+    }
+  },
+  parameters: {
+    docs: {
+      description: {
+        component: 'Component used to create skeleton effect.'
```
Comment (Miexil) [inline addon/components/o-s-s/skeleton.stories.js:64]: to create a* skeleton effect

**Review by Miexil — COMMENTED** [review]
BTW, looking at your screenshot, we will never use bars going down, what I meant with the `row` and `column` issue was more like a flex-direction.
row -> puts the skeleton bars next to each other on the same horizontal line
column -> puts the skeleton bars underneath each other on a vertical line

File: addon/components/o-s-s/skeleton.ts (hunk 2 above)
Comment (JulienVannier66) [reply addon/components/o-s-s/skeleton.ts:25]: So here `type` is also mandatory? It is not the case in the interface

Comment (aprentout) [reply addon/components/o-s-s/skeleton.ts:27]: `if (this.args.type)` prevent mandatory

**Review by Miexil — APPROVED** [review]

**Review by JulienVannier66 — APPROVED** [review]

**Review by phndiaye — APPROVED** [review]

**Pull request merged**